scripts/earth_osm.py

```diff
@@ -0,0 +1,107 @@
+
+    Input:
+        input_file: The path to the input JSON file
+    Returns:
+        output_file: The path to the output GeoJSON file.
+    """
+    # Load the custom JSON data
+    with open(input_file) as f:
+        data = json.load(f)
+
+    # Create a list of features from the nodes
+    features = []
+    for node_id, node_data in data["Data"]["Node"].items():
+        lon, lat = node_data["lonlat"]
+        geometry = {"type": "Point", "coordinates": [lon, lat]}
+        feature = {
+            "type": "Feature",
+            "id": node_id,
+            "geometry": geometry,
+            "properties": node_data["tags"],
+        }
+        features.append(feature)
+
+    # Create a FeatureCollection from the features
+    feature_collection = {"type": "FeatureCollection", "features": features}
+
+    # Write the FeatureCollection to a GeoJSON file
+    with open(output_file, "w") as f:
+        json.dump(feature_collection, f)
+
+
+def extract_points_inside_microgrids(
+    input_building_file, input_microgrid_file, output_file
+):
+    with open(input_building_file) as f:
+        points_geojson = json.load(f)
+
+    with open(input_microgrid_file) as f:
+        rectangle_geojson = json.load(f)
+
+    # Iterate over the rectangles and extract the points inside each one
+    points_inside_rectangles = []
+    for feature in rectangle_geojson["features"]:
+        rectangle_coords = feature["geometry"]["coordinates"][0]
+        rectangle_coords = [
+            (lon, lat) for lat, lon in rectangle_coords
+        ]  # Swap lat and lon
+        rectangle_polygon = Polygon(rectangle_coords)
+
+        # Extract the points inside the rectangle
+        points_in_rectangle = []
+        for point_feature in points_geojson["features"]:
+            point_coords = point_feature["geometry"]["coordinates"]
+            point_coords = (point_coords[1], point_coords[0])
+            point = Point(point_coords)
+            if rectangle_polygon.contains(point):
+                # Add the microgrid_id property to the point feature
+                point_feature["properties"]["microgrid_id"] = feature["properties"][
+                    "name"
+                ]
+                points_in_rectangle.append(point_feature)
+
+        # Add the points to the list of points inside all rectangles
+        points_inside_rectangles.extend(points_in_rectangle)
+
+    # Save the results to a new geojson file
+    result_geojson = {"type": "FeatureCollection", "features": points_inside_rectangles}
+
+    with open(output_file, "w") as f:
+        json.dump(result_geojson, f)
+
+
+if __name__ == "__main__":
+    if "snakemake" not in globals():
+        from _helpers_dist import mock_snakemake
+
+        os.chdir(os.path.dirname(os.path.abspath(__file__)))
+        snakemake = mock_snakemake("earth_osm")
+        sets_path_to_root("pypsa-distribution")
+
+    configure_logging(snakemake)
+
+    transform_json_to_geojson(
```

Comment:
So, we need a rule/script to download the raw osm data, similarly to download_osm_data in pypsa-earth and then this could be another script that filter them